internal/contract/infrastructure_machine.go

```diff
@@ -0,0 +1,112 @@
+func (m *InfrastructureMachineContract) Ready() *Bool {
+	return &Bool{
+		path: []string{"status", "ready"},
+	}
+}
+
+// FailureReason provides access to the status.failureReason field in an InfrastructureMachine object. Note that this field is optional.
+func (m *InfrastructureMachineContract) FailureReason() *String {
+	return &String{
+		path: []string{"status", "failureReason"},
+	}
+}
+
+// FailureMessage provides access to the status.failureMessage field in an InfrastructureMachine object. Note that this field is optional.
+func (m *InfrastructureMachineContract) FailureMessage() *String {
+	return &String{
+		path: []string{"status", "failureMessage"},
+	}
+}
+
+// Addresses provides access to the status.addresses field in an InfrastructureMachine object. Note that this field is optional.
+func (m *InfrastructureMachineContract) Addresses() *InfrastructureMachineAddressesList {
+	return &InfrastructureMachineAddressesList{}
+}
+
+// InfrastructureMachineAddressesList provides a helper struct for working with Addresses in a InfrastructureMachine object.
+type InfrastructureMachineAddressesList struct{}
+
+// MachineAddresses provides access to the status.addresses[i] field in an InfrastructureMachine object. Note that this field is optional.
+func (m *InfrastructureMachineAddressesList) MachineAddress(index int) *InfrastructureMachineAddress {
+	return &InfrastructureMachineAddress{}
+}
+
+// InfrastructureMachineAddresses provides a helper struct for working with Addresses in a InfrastructureMachine object.
+type InfrastructureMachineAddress struct {
+	Index int
+}
+
+// Type provides access to the status.addresses[i].type field in an InfrastructureMachine object. Note that this field is optional.
+func (m *InfrastructureMachineAddress) Type() *String {
+	return &String{
+		path: []string{"status", fmt.Sprintf("addresses[%d]", m.Index), "type"},
+	}
+}
+
+// Hostname provides access to the status.addresses[i].hostname field in an InfrastructureMachine object. Note that this field is optional.
+func (m *InfrastructureMachineAddress) Hostname() *String {
+	return &String{
+		path: []string{"status", fmt.Sprintf("addresses[%d]", m.Index), "hostname"},
+	}
+}
+
+// AddressesAlt provides access to the status.addresses field in an InfrastructureMachine object. Note that this field is optional.
+func (m *InfrastructureMachineContract) AddressesAlt() *MachineAddresses {
+	return &MachineAddresses{
+		path: []string{"status", "addresses"},
+	}
+}
+
+// ProviderID provides access to the spec.providerID field in an InfrastructureMachine object.
+func (m *InfrastructureMachineContract) ProviderID() *String {
+	return &String{
+		path: []string{"spec", "providerID"},
+	}
+
+}
+
+// FailureDomain provides access to the spec.failureDomain field in an InfrastructureMachine object. Note that this field is optional.
+func (m *InfrastructureMachineContract) FailureDomain() *String {
```

Comment:
This is not documented in https://cluster-api.sigs.k8s.io/developer/architecture/controllers/machine.html; could you. kindly open an issue to fix this as well as documenting Hostname and Addresses

Reply:
Done